src/panels/lovelace/editor/config-elements/hui-plant-status-card-editor.ts

```diff
@@ -0,0 +1,102 @@
+import { html, LitElement, PropertyDeclarations } from "@polymer/lit-element";
+import { TemplateResult } from "lit-html";
+import "@polymer/paper-input/paper-input";
+
+import { struct } from "../../common/structs/struct";
+import { EntitiesEditorEvent, EditorTarget } from "../types";
+import { hassLocalizeLitMixin } from "../../../../mixins/lit-localize-mixin";
+import { HomeAssistant } from "../../../../types";
+import { LovelaceCardEditor } from "../../types";
+import { fireEvent } from "../../../../common/dom/fire_event";
+import { Config } from "../../cards/hui-alarm-panel-card";
+import { configElementStyle } from "./config-elements-style";
+
+import "../../../../components/entity/ha-entity-picker";
+import "../../../../components/ha-icon";
+import computeDomain from "../../../../common/entity/compute_domain";
+
+const cardConfigStruct = struct({
+  type: "string",
+  entity: "string",
+  name: "string?",
+});
+
+export class HuiPlantStatusCardEditor extends hassLocalizeLitMixin(LitElement)
+  implements LovelaceCardEditor {
+  public hass?: HomeAssistant;
+  private _config?: Config;
+
+  public setConfig(config: Config): void {
+    config = cardConfigStruct(config);
+
+    this._config = { type: "plant-status", ...config };
+  }
+
+  static get properties(): PropertyDeclarations {
+    return { hass: {}, _config: {} };
+  }
+
+  get _entity(): string {
+    const entity = Object.keys(this.hass!.states).find(
```

Comment:
We should do this in the stub.

Reply:
Yeah. I noted that I my comment. Just didn't have time last night to figure it out

Reply:
I can't get this to work in the stub. and always get an undefined hass object Tried in the light card as that is Typescript to check my work and can't get around this TS error:
```
property 'hass' does not exist on type 'typeof HuiLightCard'
```